### ISSUE TITLE
coda_recognize_file_fuzzer: Segv on unknown address in coda_dynamic_type_delete

### ISSUE THREAD

**Issue by schwehr** [stcorp/coda#55]
```
==1496606==ERROR: AddressSanitizer: SEGV on unknown address (pc 0x55c53036df1e bp 0x7ffe48aa4330 sp 0x7ffe48aa4320 T0)
--
  | ==1496606==The signal is caused by a READ memory access.
  | ==1496606==Hint: this fault was caused by a dereference of a high value address (see register values below).  Dissassemble the provided pc to learn which register was used.
  | #0 0x55c53036df1e in coda_dynamic_type_delete third_party/stcorp_coda/libcoda/coda-cursor.c:160:19
  | #1 0x55c5304a9d05 in parser_info_cleanup third_party/stcorp_coda/libcoda/coda-xml-parser.c:280:13
  | #2 0x55c5304a944a in coda_xml_parse third_party/stcorp_coda/libcoda/coda-xml-parser.c:851:13
  | #3 0x55c5304a8495 in coda_xml_reopen third_party/stcorp_coda/libcoda/coda-xml.c:77:9
  | #4 0x55c530453aab in reopen_with_backend third_party/stcorp_coda/libcoda/coda-product.c:368:17
  | #5 0x55c5304508c8 in open_file third_party/stcorp_coda/libcoda/coda-product.c:550:9
  | #6 0x55c53045020a in coda_recognize_file third_party/stcorp_coda/libcoda/coda-product.c:594:9
  | #7 0x55c530366a11 in LLVMFuzzerTestOneInput third_party/stcorp_coda/fuzz/coda_recognize_file_fuzzer.cc:19
```
Trouble happens here:
```c
void coda_dynamic_type_delete(coda_dynamic_type *type)
{
    if (type == NULL)
    {
        return;
    }

    switch (type->backend)  // <-- backend is not valid
    {

```

[testcase-5067049788768256.zip](https://github.com/stcorp/coda/files/4532601/testcase-5067049788768256.zip)

**Comment by svniemeijer** [comment]
Thanks. Fixed in ce3e27804ca99891fd1ae0d17dbe362f2e23523f

**Comment by schwehr** [comment]
Verified the fix locally.
